app/models/merchant.rb

```diff
@@ -57,9 +57,8 @@ def self.revenue_date(date)
   end
 
   def self.pending_invoices(id)
-    Invoice.where(merchant_id:id)
-                    .joins(:transactions)
-                    .where("transactions.result" => "failed")
-                    .to_a.map(&:customer).uniq
+    Customer.joins(:invoices)
+            .where("invoices.merchant_id" => id)
+            .distinct.merge(Invoice.unsuccessful)
```

Comment:
Place the . on the previous line, together with the method call receiver.<br>Use 2 (not 8) spaces for indenting an expression spanning multiple lines.